File: benchmarks/csharp/LambdaPerformance.cs
using System;

public partial class LambdaPerformance : Benchmark
{
    const int ITERATIONS = 1_000_000;
    Action lambda = () => { };

    public void BenchmarkLambdaCall()
    {
        for(int i = 0; i < ITERATIONS; i++)
        {
            lambda();
        }
    }
}

File: gdextension/src/benchmarks/lambda_performance.cpp
#include "lambda_performance.h"

#include <godot_cpp/core/class_db.hpp>

using namespace godot;

void CPPBenchmarkLambdaPerformance::_bind_methods() {
	ClassDB::bind_method(D_METHOD("benchmark_lambda_call"), &CPPBenchmarkLambdaPerformance::benchmark_lambda_call);
}

void CPPBenchmarkLambdaPerformance::benchmark_lambda_call() {
	auto lambda = []() {};
	for (int i = 0; i < iterations; i++)
		lambda();
}

CPPBenchmarkLambdaPerformance::CPPBenchmarkLambdaPerformance() {}

CPPBenchmarkLambdaPerformance::~CPPBenchmarkLambdaPerformance() {}
